src/Elastic.Apm/OpenTelemetry/ElasticActivityListener.cs

```diff
@@ -0,0 +1,173 @@
+// Licensed to Elasticsearch B.V under
+// one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+#if NET5_0
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Elastic.Apm.Api;
+using Elastic.Apm.DiagnosticListeners;
+using Elastic.Apm.DistributedTracing;
+using Elastic.Apm.Helpers;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Model;
+
+namespace Elastic.Apm.OpenTelemetry
+{
+	public class ElasticActivityListener
+	{
+		internal readonly ConcurrentDictionary<string, Span> ActiveSpans = new();
+		internal readonly ConcurrentDictionary<string, Transaction> ActiveTransactions = new();
```

Comment:
Could one `ConcurrentDictionary<string, IExecutionSegment> ActiveSegments` be used to track active transactions and spans?